youtube_dl/extractor/tube8.py

```diff
@@ -1,86 +1,260 @@
 from __future__ import unicode_literals
 
+import itertools
 import re
+from time import sleep
 
+from .common import InfoExtractor
 from ..utils import (
+    traverse_obj,
+    T,
+    url_or_none,
+    parse_iso8601,
+    get_element_by_class,
+    get_element_by_id,
     int_or_none,
-    str_to_int,
+    parse_qs,
+    urljoin,
 )
-from .keezmovies import KeezMoviesIE
 
 
-class Tube8IE(KeezMoviesIE):
-    _VALID_URL = r'https?://(?:www\.)?tube8\.com/(?:[^/]+/)+(?P<display_id>[^/]+)/(?P<id>\d+)'
+class Tube8IE(InfoExtractor):
+    _VALID_URL = r'https?:\/\/(?:www\.)?tube8\.com\/+porn-video+\/(?P<id>\d+)'
     _TESTS = [{
-        'url': 'http://www.tube8.com/teen/kasia-music-video/229795/',
-        'md5': '65e20c48e6abff62ed0c3965fff13a39',
+        'url': 'https://www.tube8.com/porn-video/189530841/',
+        'md5': '532408f59e89a32027d873af6289c85a',
         'info_dict': {
-            'id': '229795',
-            'display_id': 'kasia-music-video',
+            'id': '189530841',
             'ext': 'mp4',
-            'description': 'hot teen Kasia grinding',
-            'uploader': 'unknown',
-            'title': 'Kasia music video',
-            'age_limit': 18,
-            'duration': 230,
-            'categories': ['Teen'],
-            'tags': ['dancing'],
+            'title': 'Found dildo. She let it cum in her tight ass to keep the secret',
+            'thumbnail': r're:^https?://.*\.jpg$',
+            'uploader': 'MaryKrylova',
+            'timestamp': 1718961736,
+            'upload_date': '20240621',
+        }
+    }]
+
+    def _real_extract(self, url):
```

Comment:
The Generic extractor expects a method `Tube8IE._extract_urls()`. Let's reinstate the old one in a _yt-dlp_-compatible (they can just delete the method) way.
```suggestion
    _EMBED_REGEX =  r'<iframe[^>]+\bsrc=["\'](?P<url>(?:https?:)?//(?:www\.)?tube8\.com/embed/(?:[^/]+/)+\d+)'
    
    @classmethod
    def _extract_urls(cls, webpage):
        return [m.group('url') for m in re.finditer(cls._EMBED_REGEX, webpage)]

    def _real_extract(self, url):
```
The big G sent me to https://thumbs.pro/page/3/?search=tube8 where there was at least one valid embedded Tube8 video.

Reply:
With your _EMBED_REGEX, I cloud not match with the following iframe:

```html
<iframe frameborder="0" src="https://www.tube8.com/embed/189530931/" width="560" height="340" scrolling="no" allowfullscreen></iframe>
```
The iframe is copied directly from the video player of Tube8. For reference, the original video is [https://www.tube8.com/porn-video/189530931/)](https://www.tube8.com/porn-video/189530931/).
On the thumbs.pro site ([https://thumbs.pro/page/3/?search=tube8](url)), I cloud not find any iframe.

So if we remove the last non-capturing group, it works:

```suggestion
    _EMBED_REGEX = r'<iframe[^>]+\bsrc=[\"\'](?P<url>(?:https?:)?//(?:www\.)?tube8\.com/embed/+\d+)'
    
    @classmethod
    def _extract_urls(cls, webpage):
        return [m.group('url') for m in re.finditer(cls._EMBED_REGEX, webpage)]
    
    def _real_extract(self, url):
```

Reply:
Yes, but then the embed URLs on the page that I linked aren't matched. Just tone down the repetition of the non-capturing group (and I tweaked the tag pattern too):
```suggestion
    _EMBED_REGEX =  r'<iframe [^>]*\bsrc=["\'](?P<url>(?:https?:)?//(?:www\.)?tube8\.com/embed/(?:[^/]+/)*\d+)'
    
    @classmethod
    def _extract_urls(cls, webpage):
        return [m.group('url') for m in re.finditer(cls._EMBED_REGEX, webpage)]

    def _real_extract(self, url):
```